### PR TITLE
MaintenanceOngoingException

### PR DESCRIPTION
**Pull request by rv0lt**
## Read this before submitting the PR

1. Always create a Draft PR first
2. Go through sections 1-5 below, fill them in and check all the boxes
3. Make sure that the branch is updated; if there's an "Update branch" button at the bottom of the PR, rebase or update branch.
4. When all boxes are checked, information is filled in, and the branch is updated: mark as Ready For Review and tag reviewers (top right)
5. Once there is a submitted review, implement the suggestions (if reasonable, otherwise discuss) and request an new review.

If there is a field which you are unsure about, enter the edit mode of this description or go to the [PR template](../.github/pull_request_template.md); There are invisible comments providing descriptions which may be of help.

## 1. Description / Summary

Add a filter in order to avoid logging out the Maintenace mode expcetion.

The issue is that, every time that a 500 error is generated (SERVER ERROR) it is reported as an exception in the logs in the backend (in contrast with 400 error which are not reported as such).

A maintenance mode error should be, by definition, a 503 error. Therefore, to avoid reporting it, we will filter out in the logs.

## 2. Jira task / GitHub issue

[_Link to the github issue or add the Jira task ID here._](https://scilifelab.atlassian.net/jira/software/projects/DDS/boards/13?selectedIssue=DDS-2176)

## 3. Type of change

What _type of change(s)_ does the PR contain?

**Check the relevant boxes below. For an explanation of the different sections, enter edit mode of this PR description template.**

- [X] New feature
  - [ ] Breaking: _Why / How? Add info here._ 
  - [X] Non-breaking 
- [ ] Database change: _Remember the to include a new migration version, **or** explain here why it's not needed._ 
- [ ] Bug fix 
- [ ] Security Alert fix 
  - [ ] Package update 
    - [ ] Major version update 
- [ ] Documentation 
- [ ] Workflow 
- [ ] Tests **only** 

## 4. Additional information

- [ ] [Sprintlog](../SPRINTLOG.md) 
- [ ] Blocking PRs 
  - [ ] Merged 
- [ ] PR to `master` branch: \_If checked, read [the release instructions](../doc/procedures/new_release.md) 
  - [ ] I have followed steps 1-8. 

## 5. Actions / Scans

_Check the boxes when the specified checks have passed._

**For information on what the different checks do and how to fix it if they're failing, enter edit mode of this description or go to the [PR template](../.github/pull_request_template.md).**

- [X] **Black**

- [X] **Prettier**

- [X] **Yamllint**

- [X] **Tests**

- [X] **CodeQL**

- [X] **Trivy**

- [X] **Snyk**

### CODE REVIEW

File: dds_web/__init__.py [hunk 1]
```diff
@@ -68,6 +69,20 @@
 ####################################################################################################
 
 
+class FilterMaintenanceExc(flask_logging.Filter):
+
+    def filter(record):
+        """
+        Filters log records to exclude those with MaintenanceOngoingException.
+        Returns:
+            bool: True if the log record should be processed, False if it should be filtered out.
```
Comment (valyo) [inline dds_web/__init__.py:78]:
```suggestion
            bool: True if the log record should be logged, False if it should be filtered out.
```

**Review by valyo — CHANGES_REQUESTED** [review]
Apart from the minor docstring change suggestion, I like the solution. I produces this in the container log:
`| 172.18.0.6 - - [08/Nov/2024 13:55:36] "GET /api/v1/proj/list HTTP/1.1" 503 -`
and in the actions.log:
`{"action": {"message": "Maintenance of DDS is ongoing.", "resource": "/api/v1/user/info", "project": null, "user": "None (172.18.0.6)", "response": "503 Service Unavailable", "event": "exception.MaintenanceOngoingException", "logger": "actions", "level": "warning", "timestamp": "2024-11-08T13:55:24.950844Z"}}`

**Review by valyo — APPROVED** [review]

**Comment by valyo** [comment]
@rv0lt just a reminder not to merger despite of my approval

**Review by i-oden — CHANGES_REQUESTED** [review]
Want / need to weigh in.

File: SPRINTLOG.md [hunk 2]
```diff
@@ -453,4 +453,5 @@ _Nothing merged during this sprint_
 
 # 2024-11-04 - 2024-11-15
 
+- Filter out the MaintenanceModeException from the logs ([#1573](https://github.com/ScilifelabDataCentre/dds_web/pull/1573))
```
Comment (i-oden) [inline SPRINTLOG.md:456]: Move rows to correct sprint.

**Review by i-oden — COMMENTED** [review]
I like it! Question though regarding the use of `flask.logging` as opposed to the regular `logging` library -- are there no filters etc in the standard logging library?

File: dds_web/__init__.py [hunk 3]
```diff
@@ -68,6 +69,20 @@
 ####################################################################################################
 
 
+class FilterMaintenanceExc(flask_logging.Filter):
```
Comment (i-oden) [inline dds_web/__init__.py:72]:
```suggestion
class FilterMaintenanceExc(logging.Filter):
```

File: dds_web/__init__.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 from flask_httpauth import HTTPBasicAuth, HTTPTokenAuth
 import flask_mail
 import flask_login
+from flask.logging import logging as flask_logging
```
Comment (i-oden) [inline dds_web/__init__.py:22]:
```suggestion
```

**Review by i-oden — CHANGES_REQUESTED** [review]
I asked in my previous comment if `logging` didn't have a filter, and now I have checked and `flask.logging` seems to just use the standard logging library? So I tried just using the regular logging library and it seems to be working exactly the same? So then we should be able to remove `flask.logging`, unless I'm missing something.

**Comment by rv0lt** [comment]
> I asked in my previous comment if `logging` didn't have a filter, and now I have checked and `flask.logging` seems to just use the standard logging library? So I tried just using the regular logging library and it seems to be working exactly the same? So then we should be able to remove `flask.logging`, unless I'm missing something.

I will test again your solution then!

**Review by i-oden — APPROVED** [review]
All good to merge!

**Pull request merged**